modules/hal_nordic/nrfs/dvfs/ld_dvfs_handler.c

```diff
@@ -0,0 +1,378 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "ld_dvfs_handler.h"
+#include "ld_dvfs.h"
+
+#include <hal/nrf_hsfll.h>
+#include <nrfs_dvfs.h>
+#include <nrfs_backend_ipc_service.h>
+
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(LD_DVFS_LIB, CONFIG_LOCAL_DOMAIN_DVFS_LIB_LOG_LEVEL);
+
+static K_SEM_DEFINE(dvfs_service_sync_sem, 0, 1);
+static K_SEM_DEFINE(dvfs_service_idle_sem, 0, 1);
+
+#define DVFS_SERV_HDL_INIT_DONE_BIT_POS (0)
+#define DVFS_SERV_HDL_FREQ_CHANGE_IN_PROGRESS_BIT_POS (1)
+
+static atomic_t dvfs_service_handler_state_bits;
+
+static enum dvfs_frequency_setting current_freq_setting;
+
+/**
+ * @brief Helper function to set bit in state variable.
+ *
+ * @param bit_pos bit position in state variable.
+ */
```

Comment:
Please remove redundant comments / redundant doxygen comments.

Reply:
I understand that documentation for static functions is not needed, but in the case we have it I would like to leave it as is.

Reply:
Again, please remove redundant comments / redundant doxygen comments. It does not provide valuable information.

Reply:
But what is the issue with having those comments? They don't look that bad to me

Reply:
removed the most controversial comments